### ISSUE TITLE
[FR] "Auto Import" Feature for Select and Multi-Selection Fields

### ISSUE THREAD

**Issue by Bluemoondragon07** [mdelobelle/metadatamenu#322]
Hi! Firstly, thank you so much for working on this amazing plugin. I've been using it for months, but I'm just now discovering how powerful it really is. 

I have a small idea that isn't super important or necessary, but I personally think it would be a neat addition. With select and multi-select fields, right now you have to manually define each option. But, what if there was an **AUTO** button that you press, and it scans the vault for instances of the current key, saves their values, and converts them all into options? So every value that exists for the current key in the vault automatically becomes an option.

I think it's a cool idea, but I'm not sure if it can be practically implemented. Again, thank you so much for this plug-in--it's drastically improved the quality of the app for me and a lot of other people!

**Comment by Krakor92** [comment]
What you describe can already be done using `Values returned from a dataview query` as the `Values source type` of your Multi or Select field: 

```js
dv.pages().yourProperty.distinct()
```

For a Link or MultiLink field, the syntax is different:
```js
dv.pages().map(p=>p.yourProperty)
```

Note, however, that the intelissense computation time depends on the size of your vault.
I have a thousand notes right now and don't really have problem with that so I guess it should be fine for most people

**Comment by Bluemoondragon07** [comment]
It worked! Thanks for telling me about that. 

So I guess an auto-import feature really isn't required if you can already get every existing value for a key in your vault. Especially since you can already filter results.

But how about appending extra options that aren't defined in existing notes to the query? Is it possible to do that already? If it isn't, *maybe* an auto-import feature would be more useful, but if it is possible already than yeah, auto-import isn't so important.

**Comment by Krakor92** [comment]
Why would you want to do that? If it doesn't already exist, add it manually in one of your notes. Then it will appear next to the other values when setting the property the next time

**Comment by Bluemoondragon07** [comment]
True. That's a much more efficient method. In fact, I started using that method immediately after you replied earlier.

But, just out of curiosity, is it possible?
